services/libs/aster-std/src/net/iface/util.rs

```diff
@@ -47,29 +49,28 @@ impl BindPortConfig {
 }
 
 pub fn spawn_background_poll_thread(iface: Arc<dyn Iface>) {
-    // FIXME: use timer or wait_timeout when timer is enable.
     let task_fn = move || {
-        debug!("spawn background poll thread");
+        trace!("spawn background poll thread for {}", iface.name());
+        let wait_queue = iface.polling_wait_queue();
         loop {
-            let next_poll_time = if let Some(next_poll_time) = iface.next_poll_at_ms() {
-                next_poll_time
-            } else {
-                Thread::yield_now();
+            let Some(next_poll_at_ms) = iface.next_poll_at_ms() else {
+                // Sleep for a rather long timeout
+                wait_queue.wait_timeout(&Duration::from_secs(10));
                 continue;
             };
-            let now = read_monotonic_milli_seconds();
-            if now > next_poll_time {
-                // FIXME: now is later than next poll time. This may cause problem.
+
+            let now_as_ms = read_monotonic_milli_seconds();
+
+            // FIXME: how to choose a better threshold?
+            const POLL_INTERVAL_THRESHOLD: u64 = 5;
+            if now_as_ms >= next_poll_at_ms || next_poll_at_ms - now_as_ms < POLL_INTERVAL_THRESHOLD
```

Comment:
maybe the condition can be shortened to `now_as_ms > next_poll_at_ms - POLL_INTERVAL_THRESHOLD`?

Reply:
Thanks for your comment. It is fixed.